internal/armada/repository/event_job_status.go

```diff
@@ -25,24 +38,59 @@ func (p *EventJobStatusProcessor) Start() {
 	}
 }
 
-func (p *EventJobStatusProcessor) handleMessage(eventMessage *api.EventMessage) error {
-	// TODO: batching???
-	event, err := api.UnwrapEvent(eventMessage)
+func (p *EventJobStatusProcessor) handleMessage(message *eventstream.Message) error {
+	event, err := api.UnwrapEvent(message.EventMessage)
 	if err != nil {
 		log.Errorf("error while unwrapping eventmessage: %v", err)
 		return err
 	}
 
-	switch event := event.(type) {
+	switch event.(type) {
 	case *api.JobRunningEvent:
-		err = p.jobRepository.UpdateStartTime(event.JobId, event.ClusterId, event.Created)
+		err = p.batcher.Report(message)
 		if err != nil {
-			log.Errorf("error while updating job start time: %v", err)
-			if err.Error() != JobNotFound {
-				return err
-			}
+			log.Errorf("error when reporting job status event to batcher: %v", err)
 		}
 	}
 
 	return nil
 }
+
+func (p *EventJobStatusProcessor) handleBatch(batch []*eventstream.Message) error {
+	var jobStartInfos []*JobStartInfo
+	for _, msg := range batch {
+		event, err := api.UnwrapEvent(msg.EventMessage)
+		if err != nil {
+			log.Errorf("error while unwrapping event message: %v", err)
+			return err
+		}
+		switch event := event.(type) {
+		case *api.JobRunningEvent:
+			jobStartInfos = append(jobStartInfos, &JobStartInfo{
+				JobId:     event.GetJobId(),
+				ClusterId: event.ClusterId,
+				StartTime: event.Created,
+			})
+		}
+	}
+
+	jobErrors, err := p.jobRepository.UpdateStartTime(jobStartInfos)
+	if err != nil {
+		log.Errorf("error when updating start times for jobs: %v", err)
+		return err
+	}
+	if len(jobErrors) < len(batch) {
```

Comment:
Maybe this should be != ?
If job errors > batch, we'd still get an index out of range in the loop below.

Should it be called updateResults or something?